testing/conformance/tests/runner.rs

```diff
@@ -177,125 +180,30 @@ async fn conformance_test_runner() -> anyhow::Result<()> {
         tracer.export_tombstones()?;
     }
 
-    if failed > 0 {
+    if counters.failed > 0 {
         Err(anyhow!("some vectors failed"))
     } else {
         Ok(())
     }
 }
 
-/// Runs a single test vector and returns a list of VectorResults,
-/// one per variant.
-async fn run_vector(
-    path: PathBuf,
-    stats: TestStatsRef,
-    tracer: TestTraceExporterRef,
-) -> anyhow::Result<impl Iterator<Item = impl Future<Output = anyhow::Result<VariantResult>>>> {
-    let file = File::open(&path)?;
-    let reader = BufReader::new(file);
-
-    // Test vectors have the form:
```

Comment:
I'm not understanding how all the code in this function manage to be get deleted in this refactor, what is rayon giving us that makes all of this go away? Or does the deletion come mainly from basic cleanup in the process?

Reply:
A lot of this code was duplicated from `MessageVector::from_file` (not sure why). There's also some stuff here that makes it possible to configure test-vector runs with environment variables, but we've never really used them.